src/acvp_rsa.c

```diff
@@ -236,6 +237,101 @@ static ACVP_RESULT acvp_rsa_release_tc(ACVP_RSA_TC *stc)
     return ACVP_SUCCESS;
 }
 
+static int
+acvp_kat_rsa_keygen(int info_gen_by_server, int rand_pq, ACVP_CAPS_LIST* cap,
+        unsigned char** p_rand, JSON_Object* testobj, unsigned char** q_rand,
+        unsigned char** seed, unsigned char** xp, unsigned char** xq,
+        unsigned char** xp1, unsigned char** xq1, unsigned char** xp2,
+        unsigned char** xq2)
+{
+    info_gen_by_server =
+            cap->cap.rsa_cap->rsa_cap_mode_list->cap_mode_attrs.keygen->info_gen_by_server;
+    if (!info_gen_by_server) {
+        if (rand_pq == RSA_RAND_PQ_B33) {
+            // "probRP"
+            *p_rand = (unsigned char*) json_object_get_string(testobj, "pRand");
+            *q_rand = (unsigned char*) json_object_get_string(testobj, "qRand");
+        }
+    }
+    else {
+        switch (rand_pq)
+        {
+        case RSA_RAND_PQ_B32: // "provRP"
+            *seed = (unsigned char*) json_object_get_string(testobj, "seed");
+            break;
+        case RSA_RAND_PQ_B34: // "provPC"
+            *seed = (unsigned char*) json_object_get_string(testobj, "seed");
+            break;
+        case RSA_RAND_PQ_B35: // "bothPC"
+            *seed = (unsigned char*) json_object_get_string(testobj, "seed");
+            *xp = (unsigned char*) json_object_get_string(testobj, "xP");
+            *xq = (unsigned char*) json_object_get_string(testobj, "xQ");
+            break;
+        case RSA_RAND_PQ_B36: // "probPC"
+            *seed = (unsigned char*) json_object_get_string(testobj, "seed");
+            *xp = (unsigned char*) json_object_get_string(testobj, "xP");
+            *xq = (unsigned char*) json_object_get_string(testobj, "xQ");
+            *xp1 = (unsigned char*) json_object_get_string(testobj, "xP1");
+            *xq1 = (unsigned char*) json_object_get_string(testobj, "xQ1");
+            *xp2 = (unsigned char*) json_object_get_string(testobj, "xP2");
+            *xq2 = (unsigned char*) json_object_get_string(testobj, "xQ2");
+            break;
+        case RSA_RAND_PQ_B33:
+        default:
+            break;
+        }
+    }
+    return info_gen_by_server;
+}
+
+ACVP_RESULT
+acvp_process_keygen(int rand_pq, unsigned int mod, int info_gen_by_server,
```

Comment:
Could we call this 'prepare' or 'setup' or something instead of process since the actual processing happens once its passed to the crypto handler callback method? :)